tla/consensus/ccfraft.tla

```diff
@@ -766,16 +766,16 @@ ChangeConfigurationInt(i, newConfiguration) ==
     /\ leadershipState[i] = Leader
     \* Configuration is not equal to the previous configuration.
     /\ newConfiguration /= MaxConfiguration(i)
-    \* CCF's integrity demands that a previously removed server cannot rejoin the network,
-    \* i.e., be re-added to a new configuration.  Instead, the node has to rejoin with a
-    \* "fresh" identity (compare sec 6.2, page 8, https://arxiv.org/abs/2310.11559).
-    /\ \A s \in newConfiguration: s \notin removedFromConfiguration
+    \* Nodes can only join a network once in CCF. This is enforced through the pre-consensus
+    \* join protocol that verifies the attestation of the joining node. The state machine of
+    \* the joining node never allows joining more than once.
+    /\ \A s \in (newConfiguration \ MaxConfiguration(i)): leadershipState[s] = None
```

Comment:
This check is not sufficient, because leadershipState does not change atomically, and so this is possible:

```
      ( 1 :> {n2} @@
        12 :> {n2, n5} @@
        16 :> {n1, n2, n4} @@
        18 :> {n2, n5} @@
...
```